presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java

```diff
@@ -647,24 +660,38 @@ protected Void visitCreateTable(CreateTable node, Integer indent)
             if (node.isNotExists()) {
                 builder.append("IF NOT EXISTS ");
             }
-            builder.append(node.getName())
-                    .append(" (");
+            String tableName = node.getName().getOriginalParts().stream()
+                    .map(Formatter::formatName)
+                    .collect(joining("."));
+            builder.append(tableName).append(" (\n");
 
-            Joiner.on(", ").appendTo(builder, transform(node.getElements(),
-                    element -> element.getName() + " " + element.getType()));
-
-            builder.append(")");
+            String columnList = node.getElements().stream()
+                    .map(column -> INDENT + formatName(column.getName()) + " " + column.getType())
+                    .collect(joining(",\n"));
+            builder.append(columnList);
+            builder.append("\n").append(")");
 
             if (!node.getProperties().isEmpty()) {
-                builder.append(" WITH (");
-                Joiner.on(", ").appendTo(builder, transform(node.getProperties().entrySet(),
-                        entry -> entry.getKey() + " = " + formatExpression(entry.getValue())));
-                builder.append(")");
+                builder.append("\nWITH (\n");
+                // Always output the table properties in sorted order
+                String propertyList = ImmutableSortedMap.copyOf(node.getProperties()).entrySet().stream()
+                        .map(entry -> INDENT + formatName(entry.getKey()) + " = " + entry.getValue())
+                        .collect(joining("," + "\n"));
+                builder.append(propertyList);
+                builder.append("\n").append(")");
             }
 
             return null;
         }
 
+        private static String formatName(String name)
+        {
+            if (NAME_PATTERN.matcher(name).matches()) {
+                return name;
+            }
+            return "\"" + name + "\"";
```

Comment:
The parameter name here already has doubled the double quotes. Doing this again will result in twice as many double quotes and failing test cases.